### PR TITLE
[telegram] Fix log spamming when transient network issue occurs

### PR DESCRIPTION
**Pull request by lsiepel**
Resolves: https://github.com/openhab/openhab-addons/issues/17113

Test jar (openHAB 4.2.0) https://1drv.ms/u/s!AnMcxmvEeupwjvsHyGFM4f3IfgiI-g?e=9ZPr7k

### CODE REVIEW

**Comment by rdslw** [comment]
Although in last months, prevailing error was 502, I checked older logs, and found also such example:
```
2023-06-27 03:11:25.110 [WARN ] [.openhab.binding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2023-06-27 03:11:25.251 [WARN ] [.openhab.binding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2023-06-27 03:11:25.392 [WARN ] [.openhab.binding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2023-06-27 03:11:25.533 [WARN ] [.openhab.binding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2023-06-27 03:11:25.678 [WARN ] [.openhab.binding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2023-06-27 03:11:25.819 [WARN ] [.openhab.binding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2023-06-27 03:11:25.961 [WARN ] [.openhab.binding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
```

I'm not 100% sure, however this also was cause by some cause of outage (my network or telegram).

Would current patch solve this issue also?

**Comment by lsiepel** [comment]
> I'm not 100% sure if that 429 errors also was caused by network outage or telegram outage, but 95% that was the root cause.
> 
> Would current patch solve this issue also?

429 too many requests is usually returned when you exceeed the rate limit of the corrosponding API. It is documented in the [Telegram API FAQ](https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this)
Implementing a back-off is beyond the scope if this PR.

**Comment by rdslw** [comment]
> > I'm not 100% sure if that 429 errors also was caused by network outage or telegram outage, but 95% that was the root cause.
> > Would current patch solve this issue also?
> 
> 429 too many requests is usually returned when you exceeed the rate limit of the corrosponding API. It is documented in the [Telegram API FAQ](https://core.telegram.org/bots/faq#my-bot-is-hitting-limits-how-do-i-avoid-this) Implementing a back-off / rate limiter / queue is beyond the scope if this PR.

Got it.

For future reference: Those 429s were the result of the plugin (no other connections to the api.telegram.org), probably connected to the  network issue but needs more investigation.

**Comment by rdslw** [comment]
First tests show it dont cause any problems. 
I'm now waiting for 502 error situation to observe it this fixes it.


FYI: Pure internet cutoff results in messages like below. They are probably correlated to rules trying to send message.
```
2024-08-04 09:52:27.199 [WARN ] [b.binding.telegram.internal.action.TelegramActions] Exception occured whilst sending message:java.net.SocketTimeoutException: Connect timed out
2024-08-04 09:57:21.143 [WARN ] [b.binding.telegram.internal.action.TelegramActions] Exception occured whilst sending message:java.net.SocketTimeoutException: Connect timed out
```

**Comment by rdslw** [comment]
The problem still persists, but with 429 error message. I think it shall be also included in the patch.

In addition:
1. there is no other devices using telegram api servers except from OH
2. cause of 429 is unknown, it may be on the telegram side
3. OH shall not spam log with the same error message
4. as far as I know, there were no messages to be delivered at that time from OH (3am)

Log:
```
2024-08-06T03:10:49  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:49  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:49  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:50  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:50  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:50  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:50  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:50  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:50  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:50  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:51  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:51  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
2024-08-06T03:10:51  [WARN ] [inding.telegram.internal.TelegramHandler] Telegram exception: GetUpdates failed with error_code 429 Too Many Requests: retry after 5
```

File: bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java [hunk 1]
```diff
@@ -224,23 +224,33 @@ private void handleExceptions(@Nullable TelegramException exception) {
...
+                        logger.warn("Telegram exception: {}", exception.getMessage());
+                        return;
                 }
-            }
-            if (exception.getCause() != null) { // cause is only non-null in case of an IOException
+            } else if (exception.getCause() != null) { // cause is only non-null in case of an IOException
                 cancelThingOnlineStatusJob();
                 updateStatus(ThingStatus.OFFLINE, ThingStatusDetail.COMMUNICATION_ERROR, exception.getMessage());
                 delayThingOnlineStatus();
                 return;
```
Comment (lolodomo) [inline bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java:250]: You can remove this return

**Review by lolodomo — APPROVED** [review]
LGTM
Just one minor remark.

File: bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java (hunk 1 above)
Comment (lsiepel) [reply bundles/org.openhab.binding.telegram/src/main/java/org/openhab/binding/telegram/internal/TelegramHandler.java:250]: Fixed

**Comment by lolodomo** [comment]
@lsiepel : do we still wait for additional tests before merging?

**Comment by lolodomo** [comment]
I have a doubt, is it just me who added the "additional testing preferred" by mistake ?

**Comment by lolodomo** [comment]
Yes, it looks like.

**Pull request merged**